File: android/src/com/dontflymoney/view/AccountsActivity.java
package com.dontflymoney.view;

import org.json.JSONException;
import org.json.JSONObject;

import com.dontflymoney.api.Request;
import com.dontflymoney.auth.Authentication;

import android.os.Bundle;
import android.provider.Settings.Secure;
import android.app.Activity;
import android.content.Intent;
import android.view.Menu;
import android.view.View;

public class AccountsActivity extends SmartActivity
{
	public AccountsActivity()
	{
		Init(this, R.layout.activity_accounts, R.menu.accounts);
	}
	
	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		getAccounts();
	}	
	
	public void getAccounts()
	{
		Request request = new Request("Account/List");
		
		request.AddParameter("ticket", Authentication.Get(getApplicationContext()));
		
		request.Post();
		
		if (request.IsSuccess())
		{
			JSONObject result = request.GetResult();
			
			try
			{
				handleResult(result);
			}
			catch (JSONException e)
			{
				AlertError("Activity json error: " + e.getMessage());
			}
		}
		else
		{
			AlertError(request.GetError());
		}
	}

	private void handleResult(JSONObject result) throws JSONException
	{
		if (result.has("error"))
		{
			String error = result.get("error").toString();
			
			if (error.contains("You are uninvited"))
			{
				BackToLogin();
			}
			
			AlertError(error);
		}
		else
		{
			Object data = result.get("data");
		}
	}	
	
	
	
}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/activities/AccountsActivity.kt
package com.darakeon.dfm.activities

import android.os.Bundle
import android.view.View
import com.darakeon.dfm.R
import com.darakeon.dfm.activities.base.SmartActivity
import com.darakeon.dfm.activities.objects.AccountsStatic
import com.darakeon.dfm.api.InternalRequest
import com.darakeon.dfm.uiHelpers.adapters.AccountAdapter
import com.darakeon.dfm.user.getAuth
import kotlinx.android.synthetic.main.accounts.*
import org.json.JSONObject

class AccountsActivity : SmartActivity<AccountsStatic>(AccountsStatic) {
	override fun contentView(): Int = R.layout.accounts

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		if (rotated && succeeded) {
			fillAccounts()
		} else {
			getAccounts()
		}
	}

	private fun getAccounts() {
		val request = InternalRequest(
			this, "Accounts/List", { d -> handleAccounts(d) }
		)
		request.addParameter("ticket", getAuth())
		request.post()
	}

	private fun handleAccounts(data: JSONObject) {
		static.accountList = data.getJSONArray("AccountList")
		fillAccounts()
	}

	private fun fillAccounts() {
		if (static.accountList.length() == 0) {
			main_table.visibility = View.GONE
			empty_list.visibility = View.VISIBLE
		} else {
			main_table.visibility = View.VISIBLE
			empty_list.visibility = View.GONE

			main_table.adapter = AccountAdapter(this, static.accountList)
		}
	}


}
